Console/Command/SynonymDeduplicateCommand.php

```diff
@@ -0,0 +1,182 @@
+<?php
+
+namespace Algolia\AlgoliaSearch\Console\Command;
+
+use Algolia\AlgoliaSearch\Helper\AlgoliaHelper;
+use Algolia\AlgoliaSearch\Service\IndexNameFetcher;
+use Algolia\AlgoliaSearch\Service\StoreNameFetcher;
+use Magento\Framework\App\State;
+use Magento\Framework\Console\Cli;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Store\Model\StoreManagerInterface;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class SynonymDeduplicateCommand extends AbstractStoreCommand
+{
+    public function __construct(
+        protected AlgoliaHelper         $algoliaHelper,
+        protected IndexNameFetcher      $indexNameFetcher,
+        protected State                 $state,
+        protected StoreNameFetcher      $storeNameFetcher,
+        protected StoreManagerInterface $storeManager,
+        ?string                         $name = null
+    ) {
+        parent::__construct($state, $storeNameFetcher, $name);
+    }
+
+    protected function getCommandPrefix(): string
+    {
+        return parent::getCommandPrefix() . 'synonyms:';
+    }
+
+    protected function getCommandName(): string
+    {
+        return 'deduplicate';
+    }
+
+    protected function getCommandDescription(): string
+    {
+        return "Identify and remove duplicate synonyms in Algolia";
```

Comment:
Maybe add a list of the 3 types that will be deduplicated (synonyms, placeholders and alternative corrections) and add an early warning that one-way synonyms will be deleted even before the confirmation prompt ?

Reply:
Bonne idée!